### PR TITLE
LF-4328: The user is not able to complete the soil amendment task after making changes

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

In order to override irrelevant fields in the DB, use `null` instead of `undefined`.

Jira link: https://lite-farm.atlassian.net/browse/LF-4328

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/util/task.ts [hunk 1]
```diff
@@ -119,14 +118,14 @@ export const formatSoilAmendmentTaskToFormStructure = (task: DBTask | FormTask):
 
       return {
         ...formattedTaskProduct,
-        weight_unit: isWeight ? rest.weight_unit : undefined,
-        volume_unit: !isWeight ? rest.volume_unit : undefined,
+        weight_unit: isWeight ? rest.weight_unit || undefined : undefined,
+        volume_unit: !isWeight ? rest.volume_unit || undefined : undefined,
```
Comment (kathyavini) [inline packages/webapp/src/util/task.ts:122]: If the database data is good (nulled properly), actually do we need these checks here? I missed it before, but now I'm wondering can it just be

```suggestion
        weight_unit: rest.weight_unit ?? undefined,
        volume_unit: rest.volume_unit ?? undefined,
```

and same for the other units below... or am I forgetting about a situation?

**Review by kathyavini — APPROVED** [review]
Yup, that makes total sense! Thank you for finding and fixing 🙏 

Looking at the `|| undefined : undefined` lines (a tiny bit yuck 😆 Although I understand why they are necessary!) now I'm suddenly thinking that our database data should be good enough -- units never present where they aren't meant to be -- that we can use them straight without reference to the values they are tracking?

Is that not true? It's a bit getting to the end of my thinking workday so I'm not sure 😅 But I think it should be safe!

**Review by Duncan-Brain — APPROVED** [review]
Nice!

File: packages/webapp/src/util/task.ts (hunk 1 above)
Comment (SayakaOno) [reply packages/webapp/src/util/task.ts]: I'm not sure why I did these only for units... I think it should be safe to go with your suggestion. Thank you!

**Review by kathyavini — APPROVED** [review]

**Review by Duncan-Brain — APPROVED** [review]

**Pull request merged**